diff --git a/.cargo/config.toml b/.cargo/config.toml
index 3b5b6e4..e69de29 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,5 +0,0 @@
-[source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
-[source.crates-io]
-replace-with = "debian-packages"
diff --git a/debian/rules b/debian/rules
index 63a745f..87ee796 100644
--- a/debian/rules
+++ b/debian/rules
@@ -4,15 +4,11 @@
 DH_VERBOSE = 1
 
 include /usr/share/dpkg/pkg-info.mk
-include /usr/share/rustc/architecture.mk
 
 export BUILD_MODE=release
-
-CARGO=/usr/share/cargo/bin/cargo
-
 export CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
 export DEB_HOST_RUST_TYPE DEB_HOST_GNU_TYPE
-export CARGO_HOME = $(CURDIR)/debian/cargo_home
+export CARGO_HOME = $(CURDIR)/.cargo
 
 export DEB_CARGO_CRATE=proxmox-datacenter-manager_$(DEB_VERSION_UPSTREAM)
 export DEB_CARGO_PACKAGE=proxmox-datacenter-manager
@@ -29,7 +25,8 @@ ifneq ($(DEB_DISTRIBUTION),UNRELEASED)
 	    die "ERROR: d/changelog <-> Cargo.toml version mismatch: $$v_cargo != $$v_deb\n" if $$v_cargo ne $$v_deb; exit(0); }' Cargo.toml
 endif
 
-	$(CARGO) prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
 	dh_auto_configure
 
+override_dh_auto_test:
+
 override_dh_strip:
